jqqb/rule.py

```diff
@@ -15,58 +15,68 @@ def __init__(self, rule_dict):
         self.value = rule_dict["value"]
 
     def evaluate(self, obj):
-        result = self.get_operator()(self.get_input(obj), self.get_value())
+        result = self.get_operator()(self.get_input(obj), self.get_value(obj))
         return result
 
     def inputs(self, obj):
         inputs = self.get_input(obj)
         return inputs
 
-    def values(self):
-        values = self.get_value()
+    def values(self, obj=None):
+        values = self.get_value(obj)
         return values
 
     def inspect(self, obj):
-        return self.inputs(obj), self.values(), self.evaluate(obj)
+        return self.inputs(obj), self.values(obj), self.evaluate(obj)
 
     def get_operator(self):
         return getattr(Operators, "eval_" + self.operator)
 
-    def get_input(self, obj):
-        fields = self.field.split(".")
+    def get_result(self, obj, input):
+        inputs = input.split(".")
         result = obj
-        steps = len(fields)
+        steps = len(inputs)
+
         for i in range(steps):
             last_step = i == steps - 1
             second_last_step = i == steps - 2
-            result = result.get(fields[i])
+            result = result.get(inputs[i])
+
             if (
                 second_last_step
                 and isinstance(result, list)
                 and isinstance(result[0], dict)
             ):
-                result = [x[fields[steps - 1]] for x in result]
+                result = [x[inputs[steps - 1]] for x in result]
+                result = result[0]
                 break
+
             result = (
                 result[0]
-                if (
-                    result is not None
-                    and isinstance(result, list)
-                    and not last_step
-                )
+                if (result is not None and isinstance(result, list) and not last_step)
                 else result
             )
             if result is None:
                 break
+        return result
+
+    def get_input(self, obj):
+        result = self.get_result(obj, self.field)
         if isinstance(result, list):
             return list(map(lambda x: self.typecast_value(x), result))
         else:
             return self.typecast_value(result)
 
-    def get_value(self):
-        if isinstance(self.value, list):
-            return list(map(lambda x: self.typecast_value(x), self.value))
-        return self.typecast_value(self.value)
+    def get_value(self, obj=None):
+        result = None
+        if self.value is not None:
+            result = self.get_result(obj, self.value)
```

Comment:
If I get this example, I think it will fail:
- I want to check the object got an empty list for a specific key
- `self.field = "test"`
- `self.value = []`

You should get an exception, because a list does not have `split` method.

Reply:
I guess `self.value` could be:
- either a custom field, example : self.value = `"actor.custom_fields.code_contrat"`
- or a value entered by the user, example   self.value = `"farine"`

In both case, we are dealing with string